project-api/ballerina-projects/src/main/java/io/ballerina/projects/writers/BaloWriter.java

```diff
@@ -29,18 +59,347 @@
+        // create the module directory in zip
+        Path packageInBalo = root.resolve(MODULES);
+        Files.createDirectory(packageInBalo);
+
+        // add default module
+        Path defaultPkgDirInBalo = packageInBalo.resolve(defaultPackageName);
+        Files.createDirectory(defaultPkgDirInBalo);
+
+        // copy resources directory
+        copyResourcesDir(packageSourceDir, defaultPkgDirInBalo);
+        // only add .bal files in the package root
+        copyBallerinaSrcFiles(packageSourceDir, defaultPkgDirInBalo);
+
+        // add other modules
+        File modulesSourceDir = new File(String.valueOf(packageSourceDir.resolve(MODULES)));
+        File[] directoryListing = modulesSourceDir.listFiles();
+
+        if (directoryListing != null) {
+            for (File moduleDir : directoryListing) {
+                if (moduleDir.isDirectory()) {
+                    // add module
+                    Path moduleDirInBalo = packageInBalo.resolve(defaultPackageName + "." + moduleDir.getName());
+                    Files.createDirectory(moduleDirInBalo);
+
+                    // copy resources directory
+                    copyResourcesDir(moduleDir.toPath(), moduleDirInBalo);
+                    // only add .bal files in the module root
+                    copyBallerinaSrcFiles(moduleDir.toPath(), moduleDirInBalo);
+                }
+            }
+        }
+    }
+
+    private static void copyResourcesDir(Path sourceDir, Path sourceDirInBalo) throws IOException {
+        Path resourcesDir = sourceDirInBalo.resolve("resources");
+        Path resourcesSrcDir = sourceDir.resolve("resources");
+
+        // if resources not exists ignore copying
+        if (resourcesSrcDir.toFile().exists()) {
+            File[] resourceFiles = new File(String.valueOf(resourcesSrcDir)).listFiles();
+            if (resourceFiles != null && resourceFiles.length > 0) {
+                Files.walkFileTree(resourcesSrcDir, new CopyResources(resourcesSrcDir, resourcesDir));
+            }
+        }
+    }
+
+    private static void copyBallerinaSrcFiles(Path sourceDirPath, Path targetPath) throws IOException {
+        File[] defaultModuleFiles = new File(String.valueOf(sourceDirPath)).listFiles();
+        if (defaultModuleFiles != null) {
+            for (File file : defaultModuleFiles) {
+                if (file.isFile() && file.getName().endsWith(".bal")) {
+                    Files.copy(file.toPath(), targetPath.resolve(file.getName()), StandardCopyOption.REPLACE_EXISTING);
+                }
+            }
+        }
+    }
+
+    private static void addPlatformLibs(Path root, Path projectDirectory, BallerinaToml ballerinaToml)
+            throws IOException {
+        //If platform libs are defined add them to balo
+        List<Library> platformLibs = ballerinaToml.getPlatform().libraries;
+        if (platformLibs == null) {
+            return;
+        }
+        Path platformLibsDir = root.resolve("lib");
```

Comment:
Shall we move `"lib"` to ProjectConstants?

Reply:
Added `LIB_DIR` to `ProjectConstants`